server.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+const net = require('net');
+const server = net.createServer();
+const parseMessage = require('./lib/parse-message.js');
+
+let sockets = [];
+
+server.on('connection', function(socket){
+  console.log('a client connected');
+  socket.write('welcome to wack chat');
+  socket.username = `user_${Math.floor(Math.random()*100)}`;
+  sockets.push(socket);
+
+  socket.on('data', function(buffer){
+    console.log(buffer.toString());
+    let message = buffer.toString();
+    if(message.startsWith('/nick'))
+      return parseMessage.nickCommand(message, socket);
+
+    if(message.startsWith('/dm'))
+      return parseMessage.dmCommand(message, sockets);
+
+    if(message.startsWith('/users'))
+      return parseMessage.users(sockets, socket);
+
+    if(message.startsWith('/troll'))
+      return parseMessage.trolololol(message, sockets);
+
+    if(message.startsWith('/ban')) {
+      return parseMessage.ban(message, sockets);
+
+    }
+
+    sockets.forEach(s => {
```

Comment:
This ``forEach`` method could actually be taken out completely because that just broadcasts anything you type to all users by default, and that's what the ``/all`` command should be doing.